In [1]:
import pandas as pd
import numpy as np

In [2]:
from pathlib import Path

In [3]:
!pwd

/Users/haohe/Desktop/Heart_shenzhen


In [4]:
root = '/Users/haohe/Desktop/Heart_shenzhen'
audio_data = root + '/audio'
data = root + '/data'

In [5]:
df = pd.read_excel(data+'/train_mapping.xlsx')

# Process valid_test

In [9]:
df_valid_test = pd.read_excel(data+'/valid_test_mapping.xlsx')
df_valid_test.head()

ComParE2018_Heartbeat.confidential  \
0                          file_name   
1                     devel_0001.wav   
2                     devel_0002.wav   
3                     devel_0003.wav   
4                     devel_0004.wav   

                                          Unnamed: 1  
0                                     orig_file_name  
1  1422900/Ausculation_Report_Eko_Recording_13168...  
2  1423640/Ausculation_Report_Eko_Recording_13169...  
3  633418/Ausculation_Report_Eko_Recording_133855...  
4  1196812/Ausculation_Report_Eko_Recording_14173...

In [10]:
df_valid_test = df_valid_test.rename(columns={
    'ComParE2018_Heartbeat.confidential': 'filename',
    'Unnamed: 1': 'orig_filename'
})

In [13]:
df_valid_test = df_valid_test.iloc[1:,:]
df_valid_test.head()

filename                                      orig_filename
1  devel_0001.wav  1422900/Ausculation_Report_Eko_Recording_13168...
2  devel_0002.wav  1423640/Ausculation_Report_Eko_Recording_13169...
3  devel_0003.wav  633418/Ausculation_Report_Eko_Recording_133855...
4  devel_0004.wav  1196812/Ausculation_Report_Eko_Recording_14173...
5  devel_0005.wav  1109541/Ausculation_Report_Eko_Recording_13871...

In [7]:
filename = df.iloc[1:,0].tolist()

In [12]:
filename[0:10]

['train_0001.wav',
 'train_0002.wav',
 'train_0003.wav',
 'train_0004.wav',
 'train_0005.wav',
 'train_0006.wav',
 'train_0007.wav',
 'train_0008.wav',
 'train_0009.wav',
 'train_0010.wav']

In [8]:
orig_filename = df.iloc[1:,1].tolist()

In [13]:
df_raw = pd.DataFrame(list(zip(filename,orig_filename)),columns=['filename','orig_filename'])

In [14]:
df_raw.head()

filename                                      orig_filename
0  train_0001.wav  1428734/Ausculation_Report_Eko_Recording_13641...
1  train_0002.wav  1430457/Ausculation_Report_Eko_Recording_14172...
2  train_0003.wav  1426832/Ausculation_Report_Eko_Recording_13592...
3  train_0004.wav  1427474/Ausculation_Report_Eko_Recording_13644...
4  train_0005.wav  648358/Ausculation_Report_Eko_Recording_136362...

In [19]:
df_raw['orig_filename'][0].split('/')[0]

'1428734'

In [81]:
df_raw['id'] = df_raw['orig_filename'].apply(lambda o: int(o.split('/')[0]))

In [82]:
df_raw = df_raw.sort_values('id')

In [83]:
df_raw.head(n=10)

filename                                      orig_filename      id
42   train_0043.wav  524215/Ausculation_Report_Eko_Recording_140111...  524215
208  train_0209.wav  524215/Ausculation_Report_Eko_Recording_140114...  524215
199  train_0200.wav  524215/Ausculation_Report_Eko_Recording_140112...  524215
200  train_0201.wav  524215/Ausculation_Report_Eko_Recording_140113...  524215
158  train_0159.wav  615897/Ausculation_Report_Eko_Recording_141773...  615897
56   train_0057.wav  615897/Ausculation_Report_Eko_Recording_141775...  615897
288  train_0289.wav  615897/Ausculation_Report_Eko_Recording_141774...  615897
57   train_0058.wav  615897/Ausculation_Report_Eko_Recording_141776...  615897
10   train_0011.wav  648358/Ausculation_Report_Eko_Recording_136364...  648358
111  train_0112.wav  648358/Ausculation_Report_Eko_Recording_136365...  648358

In [45]:
df_label = pd.read_excel(data+'/train_label.xlsx')
df_label.head()

Unnamed: 0 Unnamed: 1   Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5  \
0    住院号/门诊号         姓名         英文姓名         性别         民族         籍贯   
1         11         冯建    Feng,Jian        NaN        NaN        NaN   
2         12        NaN  Xu, Jianhua        NaN        NaN        NaN   
3     524215        叶思思     Ye, Sisi        1=男       1=汉族         温州   
4     541646        余贤地  Yu, Xiandi         1=男       1=汉族         文成   

  Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9 Unnamed: 10      Unnamed: 11  \
0      年龄（岁）     身高（cm）     体重（kg）      体温（℃）     心率（bpm）  促甲状腺素TSH（mIU/L）   
1        NaN        NaN        NaN        NaN         NaN              NaN   
2        NaN        NaN        NaN        NaN         NaN              NaN   
3         35        164         64       36.5          80             1.12   
4         62        162         65       36.3          88              0.9   

    Unnamed: 12   Unnamed: 13  
0  游离T4（pmol/L）  游离T3（pmol/L）  
1           NaN           NaN  
2           NaN           NaN  
3          8.32           4.6  
4            11           4.6

In [51]:
df_label = df_label.rename(columns={'Unnamed: 0':'id',
                         'Unnamed: 1':'name',
                         'Unnamed: 2':'eng_name',
                         'Unnamed: 3':'gender',
                         'Unnamed: 4':'cul',
                         'Unnamed: 5':'hometown',
                         'Unnamed: 6':'age',
                         'Unnamed: 7':'height',
                         'Unnamed: 8':'weight',
                         'Unnamed: 9':'temp',
                         'Unnamed: 10':'bpm',
                        'Unnamed: 11':'tsh',
                        'Unnamed: 12':'t4',
                        'Unnamed: 13':'t3'})

In [72]:
df_label = df_label.iloc[1:,:]

In [73]:
df_label.to_csv(data+'/train_label.csv')

In [48]:
df_raw.to_csv(data+'/train_mapping.csv')

In [54]:
df_label[df_label['id']==1108990]

id name        eng_name gender   cul hometown age height weight  \
23  1108990  陈成绿  Chen, Chenglv     1=男  1=汉族       苍南  74    169     70   

    temp bpm   tsh     t4   t3  
23  36.6  64  0.94  12.23  4.5

In [61]:
df_label[df_label['id'] == 1108990]['tsh'].item()

0.94

In [86]:
df_merge = df_raw.merge(df_label,on='id',how='left')

In [90]:
df_merge.to_csv(data+'/raw_all.csv',index=False)

In [91]:
df_train = df_merge[df_merge['tsh'].notna()]

In [92]:
df_train.to_csv(data+'/train.csv',index=False)

In [93]:
df_train.head()

filename                                      orig_filename      id  \
0  train_0043.wav  524215/Ausculation_Report_Eko_Recording_140111...  524215   
1  train_0209.wav  524215/Ausculation_Report_Eko_Recording_140114...  524215   
2  train_0200.wav  524215/Ausculation_Report_Eko_Recording_140112...  524215   
3  train_0201.wav  524215/Ausculation_Report_Eko_Recording_140113...  524215   
4  train_0159.wav  615897/Ausculation_Report_Eko_Recording_141773...  615897   

  name    eng_name gender   cul hometown age height weight  temp bpm   tsh  \
0  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
1  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
2  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
3  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
4  叶晓霞  Ye,Xiaoxia    2=女  1=汉族       永嘉  69    156     50  36.9  60  2.53   

      t4   t3  
0   8.32  4.6  
1   8.32  4.6  
2   8.32  4.6  
3   8.32  4.6  
4  12.34  4.2

In [95]:
len(df_train.id.unique())

77

In [96]:
df_train.shape

(394, 16)

# Fin

23 of the training data doesn't have label

total 77 unique person and 394 recordings are avaiable